File: src/main/java/com/icloud/modules/unimall/entity/UnimallSupplier.java
package com.icloud.modules.unimall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.math.BigDecimal;
import java.io.Serializable;
import java.util.Date;

/**
 * 商户表 商户表
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-05-16 10:38:18
 */
@Data
@TableName("t_unimall_supplier")
public class UnimallSupplier implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /* id */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /* 联系人带电话 */
       @TableField("phone")
       private String phone;
   	   	   /* 联系人 */
       @TableField("contact")
       private String contact;
   	   	   /* 店铺名称 */
       @TableField("shop_name")
       private String shopName;
   	   	   /* 店铺图标 */
       @TableField("shop_icon")
       private String shopIcon;
   	   	   /* appid */
       @TableField("appid")
       private String appid;
   	   	   /* appsecret */
       @TableField("appsecret")
       private String appsecret;
   	   	   /* 经度 */
       @TableField("longitude")
       private BigDecimal longitude;
   	   	   /* 纬度 */
       @TableField("latitude")
       private BigDecimal latitude;
   	   	   /* 营业时间 */
       @TableField("business_hours")
       private String businessHours;
   	   	   /* openid */
       @TableField("openid")
       private String openid;
   	   	   /* 状态 */
       @TableField("status")
       private Integer status;
   	   	   /* 配送范围 */
       @TableField("delivery_scope")
       private Integer deliveryScope;
   	   	   /* 省 */
       @TableField("province")
       private String province;
   	   	   /* 市 */
       @TableField("city")
       private String city;
   	   	   /* 县 */
       @TableField("county")
       private String county;
   	   	   /* 详细店铺地址 */
       @TableField("address")
       private String address;
   	   	   /* 排序 */
       @TableField("sort_num")
       private Integer sortNum;
   	   	   /* 创建时间 */
       @TableField("gmt_create")
       private Date gmtCreate;
   	   	   /* 更新时间 */
       @TableField("gmt_update")
       private Date gmtUpdate;
   	
}

File: src/main/resources/statics/js/modules/unimall/unimallsupplier.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'unimall/unimallsupplier/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '联系人带电话', name: 'phone', index: 'phone', width: 80 }, 			
			{ label: '联系人', name: 'contact', index: 'contact', width: 80 }, 			
			{ label: '店铺名称', name: 'shopName', index: 'shop_name', width: 80 }, 			
			{ label: '店铺图标', name: 'shopIcon', index: 'shop_icon', width: 80 }, 			
			{ label: 'appid', name: 'appid', index: 'appid', width: 80 }, 			
			{ label: 'appsecret', name: 'appsecret', index: 'appsecret', width: 80 }, 			
			{ label: '经度', name: 'longitude', index: 'longitude', width: 80 }, 			
			{ label: '纬度', name: 'latitude', index: 'latitude', width: 80 }, 			
			{ label: '营业时间', name: 'businessHours', index: 'business_hours', width: 80 }, 			
			{ label: 'openid', name: 'openid', index: 'openid', width: 80 }, 			
			{ label: '状态', name: 'status', index: 'status', width: 80 }, 			
			{ label: '配送范围', name: 'deliveryScope', index: 'delivery_scope', width: 80 }, 			
			{ label: '省', name: 'province', index: 'province', width: 80 }, 			
			{ label: '市', name: 'city', index: 'city', width: 80 }, 			
			{ label: '县', name: 'county', index: 'county', width: 80 }, 			
			{ label: '详细店铺地址', name: 'address', index: 'address', width: 80 }, 			
			{ label: '排序', name: 'sortNum', index: 'sort_num', width: 80 }, 			
			{ label: '创建时间', name: 'gmtCreate', index: 'gmt_create', width: 80 }, 			
			{ label: '更新时间', name: 'gmtUpdate', index: 'gmt_update', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		unimallSupplier: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.unimallSupplier = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.unimallSupplier.id == null ? "unimall/unimallsupplier/save" : "unimall/unimallsupplier/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.unimallSupplier),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "unimall/unimallsupplier/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "unimall/unimallsupplier/info/"+id, function(r){
                vm.unimallSupplier = r.unimallSupplier;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});